internal/instance/resource_instance.go

```diff
@@ -555,6 +534,11 @@ func (r InstanceResource) Read(ctx context.Context, req resource.ReadRequest, re
 	resp.Diagnostics.Append(diags...)
 }
 
+// Update updates the instance in the following order:
+// - Ensure instance state (stopped/running)
+// - Update configuration (config, limits, devices, profiles)
```

Comment:
Yes and no. First of all, previously we did not care in what state the instance is, since all operations could be done regardless if instance is stopped or running.
If it happened that we are updating config entries that can be update only if instance is stopped, but instance was running, an error would occur.

However, we were not able to start instance again. So, allowing users to ensure instance is running after terraform apply seems like a more valuable addition compared to handling those special cases.

We could ensure the instance is running just before the exec commands, but this still does not guarantee the entries will be updated (because we do not know whether instance was running before or not).

To very simple workaround would be to stop the instance - update the instance - and finally start the instance again. While this would ensure all config entries are always updated, in most cases this would be a no-go, since we can bring down an entire cluster in the same time just to update a few special config entries.

So, I propose that for an initial implementation of `running` attribute we just ensure the instance is in the desired state at the beginning of an update. If we later recognize the need for these special cases, we can think through and improve this mechanism (maybe by adding an attribute `stop_during_update` or something like this).